src/bitops.c

```diff
@@ -428,8 +428,8 @@ int getBitOffsetFromArgument(client *c, robj *o, size_t *offset, int hash, int b
     /* Adjust the offset by 'bits' for #<offset> form. */
     if (usehash) loffset *= bits;
 
-    /* Limit offset to 512MB in bytes */
-    if ((loffset < 0) || ((unsigned long long)loffset >> 3) >= (512*1024*1024))
+    /* Limit offset to server.proto_max_bulk_len (512MB in bytes by default) */
+    if ((loffset < 0) || (loffset >> 3) >= server.proto_max_bulk_len)
```

Comment:
i guess this should be `> server.proto_max_bulk_len`

Reply:
i'm not sure it has to be exact on the byte.

just to be sure we're not missing some off by one issue,
let's try to think of an edge case, setting proto_max_bulk_len to 1.
we would be able to do `SET x x` but not `set x xx` (it'll error when size > max).
and would be able to do `SETRANGE x 0 x` and `SETRANGE x 1 x` (it'll error when offset+len > max).
so here, `SETBIT 7 1` should pass (byte index of 0 is length of 1), and `SETBIG 8 1` (byte offset of 1 is length of 2) should fail.

so it actually looks like the code is valid, right?

Reply:
right, indeed. i get it, thanks for the explanation!